src/test/connection.test.ts

```diff
@@ -1,154 +1,149 @@
 import { assert, assertEquals } from './asserts.ts';
-import { IConnection, ITransport } from '../types.ts';
-import { FnsBag } from '../types-bag.ts';
-import { makeLocalTransportPair } from '../transport-local.ts';
-
-import { sleep } from '../util.ts';
+import { makeObservedMethods } from './test-methods.ts';
 import { EventLog } from './event-log.ts';
+import { scenarios } from './scenarios.ts';
+import { ITransportScenario } from './scenario-types.ts';
 
 //================================================================================
 
-const makeObservedMethods = (e: EventLog) => {
-    return {
-        add: (a: number, b: number) => {
-            e.observe(`$${a} + ${b} = ${a + b}`);
-            return a + b;
-        },
-        addAsync: async (a: number, b: number) => {
-            await sleep(10);
-            e.observe(`async: $${a} + ${b} = ${a + b}`);
-            await sleep(10);
-            return a + b;
-        },
-        shout: (s: string) => {
-            e.observe('shout: ' + s.toLocaleUpperCase());
-        },
-        shoutAsync: async (s: string) => {
-            await sleep(10);
-            e.observe('shoutAsync: ' + s.toLocaleUpperCase());
-            await sleep(10);
-        },
-        alwaysError: (msg: string) => {
-            e.observe(`alwaysError: ${msg}`);
-            throw new Error(msg);
-        },
-    };
-};
-
 async function testConnectionNotify(
     t: Deno.TestContext,
-    conn: IConnection<ReturnType<typeof makeObservedMethods>>,
+    scenario: ITransportScenario<ReturnType<typeof makeObservedMethods>>,
     e: EventLog,
 ) {
-    await t.step('notify', async () => {
-        e.clear();
-        e.note('about to shout');
-        await conn.notify('shout', 'hello');
-        e.expect('shout: \'HELLO\'');
-        e.note('shouted');
-        await conn.notify('shoutAsync', 'hello');
+    await t.step({
+        name: `notify + ${scenario.name}`,
+        // There may be async ops on the client and server
+        sanitizeOps: false,
```

Comment:
Hopefully we can resolve these async ops, let's not forget about them, but ok to merge for now